File: data/code_pairs/cpp/1408.cpp
#include <cassert>
#include <iostream>

int main() {
    int i;
    float r;
    bool enteredLoop = false;

    // This loop will not run because the condition is never true.
    for (i = 1; i <= 0; ++i) {
        enteredLoop = true;
    }
    assert(!enteredLoop && "Loop with integer counter from 1 to 0 unexpectedly ran.");

    enteredLoop = false; // Reset for next test
    // This loop's condition is never true, so it will also not run.
    for (i = 1; i <= -1; i += 1) {
        enteredLoop = true;
    }
    assert(!enteredLoop && "Loop with integer counter from 1 to -1 with step 1 unexpectedly ran.");

    enteredLoop = false; // Reset for next test
    // This loop's condition is never true because start < end and step is negative.
    for (i = 1; i >= 2; i -= 1) {
        enteredLoop = true;
    }
    assert(!enteredLoop && "Loop with integer counter from 1 to 2 with step -1 unexpectedly ran.");

    // Loops with a step of 0 are undefined in Fortran and not directly translatable to safe C++ code due to potential for infinite loops.

    std::cout << "All tests passed." << std::endl;
    return 0;
}

File: data/code_pairs/fortran/1408.f90
program test_loops
    implicit none
    integer :: i
    real :: r

    ! Note: Fortran loops with step 0 or conditions that are never true essentially "do not run".

    do i = 1, 0
    end do
    print *, "Loop with integer counter from 1 to 0 did not run."

    do i = 1, -1, 1
    end do
    print *, "Loop with integer counter from 1 to -1 with step 1 did not run."

    do i = 1, 2, -1
    end do
    print *, "Loop with integer counter from 1 to 2 with step -1 did not run."

    ! Loops with steps of 0 are undefined in Fortran; they are not represented here.

    do r = 1, 0
    end do
    print *, "Loop with real counter from 1 to 0 did not run."

    do r = 1, -1, 1
    end do
    print *, "Loop with real counter from 1 to -1 with step 1 did not run."

    do r = 1, 2, -1
    end do
    print *, "Loop with real counter from 1 to 2 with step -1 did not run."

    ! Loops with steps of 0 are undefined in Fortran; they are not represented here.

end program test_loops